Backport missing Gutenberg tests to Core

anton-vlasenko opened the pull request:
<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

This PR aims to backport `WP_Navigation_Test` and `WP_Global_Styles_Test` test classes to Core.
These tests haven't been backported to Core.
Original tests classes:

1. https://github.com/WordPress/gutenberg/blob/release/13.6/phpunit/global-styles-test.php
2. https://github.com/WordPress/gutenberg/blob/release/13.6/phpunit/navigation-test.php 

Trac ticket: https://core.trac.wordpress.org/ticket/56266

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

anton-vlasenko committed:
Backport test for wp_filter_global_styles_post function (Gutenberg to Core).

anton-vlasenko committed:
Backport tests for _disable_block_editor_for_navigation_post_type, _disable_content_editor_for_navigation_post_type, _enable_content_editor_for_navigation_post_type functions from Gutenberg to Core.

anton-vlasenko committed:
Refactor test_wp_filter_global_styles_post:
1. Rename it to test_wp_filter_global_styles_post_returns_correct_value.
2. Better validation message.

anton-vlasenko committed:
Fix code style.

anton-vlasenko committed:
Add access modifier to test_wp_filter_global_styles_post_returns_correct_value.

hellofromtonya commented on tests/phpunit/tests/editor/disableBlockEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,27 @@
+
+	const NAVIGATION_POST_TYPE     = 'wp_navigation';
+	const NON_NAVIGATION_POST_TYPE = 'wp_non_navigation';
+
```
In the original test, it was enabling the editor support in setUp. Is this needed in this test? Does seem to be as the test is passing. But asking to ensure.

https://github.com/WordPress/gutenberg/blob/release/13.6/phpunit/navigation-test.php#L12-L18

```php
	public function setUp() {
		$this->enable_editor_support();
	}

	public function tearDown() {
		$this->enable_editor_support();
	}
```

anton-vlasenko replied:
I can't recall why it was implement this way in the original class.
But we have to restore editor support for navigation blocks. `Tests_Editor_DisableContentEditorForNavigationPostType` and `Tests_Editor_EnableContentEditorForNavigationPostType` classes disable it for testing purposes. IMO it should be enabled back not to affect other tests.

costdev replied:
Looking at `_disable_block_editor_for_navigation_post_type()`, it's just a check for `'wp_navigation'` and isn't currently affecting other tests or being affected by other tests.

```php
function _disable_block_editor_for_navigation_post_type( $value, $post_type ) {
	if ( 'wp_navigation' === $post_type ) {
		return false;
	}

	return $value;
}
```

anton-vlasenko replied:
Yes, `_disable_block_editor_for_navigation_post_type ` is kind of useless (if I may say so), but in my opinion we should still have unit tests for it since it's in Core now.
I would like to hear your suggestion. Any ideas on how to improve this unit test, @costdev ?

costdev replied:
- Merging the two test methods and using a data provider would make it easier to cover the four main paths, and if more post types are included in future, for example, more datasets can just be added to the provider.
- The assertion would change to `assertSame()` so that a `(bool) $expected` parameter can be passed.
- In addition, I would pass `(bool) $value` and `(string) $post_type` arguments to the test method.

There would be four datasets at the moment:
```php
'non-navigation post type and true'  => array(
    'post_type' => static::NON_NAVIGATION_POST_TYPE,
    'value'     => true,
    'expected'  => true,
),
'non-navigation post type and false' => array(
    'post_type' => static::NON_NAVIGATION_POST_TYPE,
    'value'     => false,
    'expected'  => false,
),
'navigation post type and true'      => array(
    'post_type' => static::NAVIGATION_POST_TYPE,
    'value'     => true,
    'expected'  => false,
),
'navigation post type and false'     => array(
    'post_type' => static::NAVIGATION_POST_TYPE,
    'value'     => false,
    'expected'  => false,
),
```

I think that's all for the _unit_ test coverage of this function and that should cover the current implementation.

In future, we may find that integration tests could improve coverage using datasets for all registered post types to ensure there are no edge cases. This would use either `get_post_types()`, or the `$wp_post_types` global directly.

anton-vlasenko replied:
Thank you for the feedback, @costdev.
I thought it would be enough if the test covered all lines of the function. However, I agree that the data provider is better for covering edge cases.
Fixed in 9aa06d424028af9187d41fc45b89e0b8ec2dfac2

hellofromtonya commented on tests/phpunit/tests/editor/disableContentEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,54 @@
+		return post_type_supports( static::NAVIGATION_POST_TYPE, 'editor' );
+	}
+
+	private function create_post( $type ) {
+		$post            = new WP_Post( new StdClass() );
+		$post->post_type = $type;
```
Suggestion: Change `$type` to `$post_type` for self-documenting and improving readability.

anton-vlasenko replied:
Fixed in aa0a54b55834d2bf425ea9fbc4040f1df7169518

hellofromtonya commented on tests/phpunit/tests/editor/enableContentEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,72 @@
+		return post_type_supports( static::NAVIGATION_POST_TYPE, 'editor' );
+	}
+
+	private function create_post( $type ) {
+		$post            = new WP_Post( new StdClass() );
+		$post->post_type = $type;
```
Suggestion: Change `$type` to `$post_type` for self-documenting and improving readability.

anton-vlasenko replied:
Fixed in d50321de684779feb49bc6aef11a9f7cd654e2ad

hellofromtonya commented on tests/phpunit/tests/kses.php:
```diff
@@ -1991,4 +1991,28 @@ public function data_kses_globals_are_defined() {
+ 		}';
+
+		$filtered_user_theme_json = wp_filter_global_styles_post( $user_theme_json );
+		$this->assertEquals( $user_theme_json, $filtered_user_theme_json, 'Filtered and expected json data must match.' );
```
Use `assertSame()` instead of `assertEquals()`.

hellofromtonya replied:
As there are no other tests for this function, I'd also suggest moving this function's test to its own test class: `tests/phpunit/tests/kses/wpFilterGlobalStylesPost.php`. In doing so, it will kick-off the work to separate each kses function into its own test class.

anton-vlasenko replied:
Fixed in ee805ff646fbe5df325ff6c57ab2a0c99ce26761

anton-vlasenko replied:
Good point. Fixed in 895f6cfdd799ee50133e2075d506c464f9b5cda2

anton-vlasenko committed:
Disable editor support at a test method level.
IMO this should be decided at a test method level.
Not all tests may need to disable editor support in the future.

anton-vlasenko committed:
Rename $post to $post_type (a better name for this variable).

anton-vlasenko committed:
Rename $post to $post_type (a better name for this variable).

anton-vlasenko committed:
We should restore the state after each run.

anton-vlasenko committed:
Refactor Tests_Kses::Tests_Kses
Move this test method to a separate class.

costdev commented on tests/phpunit/tests/editor/disableContentEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,63 @@
+	 */
+	public function test_it_doesnt_disable_content_editor_for_non_navigation_type_posts() {
+		$post = $this->create_non_navigation_post();
+		$this->assertTrue( $this->supports_block_editor() );
```
All test methods in the PR containing multiple assertions need a `message` parameter added to each assertion. [[Ref]](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#using-assertions)

anton-vlasenko replied:
Yes, we discussed error messages some time ago. I didn't notice that for some reason.
Thank you for pointing it out.
Fixed in 8da8362eb2ae2083d2d5e5e81cfa454526442005

costdev commented on tests/phpunit/tests/editor/disableContentEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,63 @@
+	}
+
+	private function create_post( $post_type ) {
+		$post            = new WP_Post( new StdClass() );
```
```suggestion
		$post            = new WP_Post( new stdClass() );
```
1. While class names are case-insensitive, for consistency and ease of searching, we should use the class names as they are named in the PHP manual.
2. Question: Should this be a `WP_Post`, or should it use the `$this->factory()->post->create()` method? I'm not sure what the answer is here.

anton-vlasenko replied:
1. My bad. Fixed in https://github.com/WordPress/wordpress-develop/pull/3005/commits/add6a919ddb97162e906d4821b269c9e0b856dae
2. Yes, I'm aware of `$this->factory()->post->create()`. In this particular case, posts don't have to be saved to the database (that was the main reason not to use `WP_UnitTest_Factory_For_Post::create`), but I don't mind using the factory as long as tests pass. Refactored in [9d26fc7](https://github.com/WordPress/wordpress-develop/commit/9d26fc76c5277641c05185e54adffcf7e4b1c2b5)

costdev commented on tests/phpunit/tests/editor/enableContentEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,69 @@
+	}
+
+	private function create_post( $post_type ) {
+		$post            = new WP_Post( new StdClass() );
```
```suggestion
		$post            = new WP_Post( new stdClass() );
```
1. While class names are case-insensitive, for consistency and ease of searching, we should use the class names as they are named in the PHP manual.
2. Question: Should this be a `WP_Post`, or should it use the `$this->factory()->post->create()` method? I'm not sure what the answer is here.

anton-vlasenko replied:
https://github.com/WordPress/wordpress-develop/pull/3005/files#r928870956

costdev reviewed (changes requested):
@anton-vlasenko Just a few minor suggested changes and a question regarding the use of `WP_Post` vs `$this->factory()->post_create()`.

anton-vlasenko committed:
Fix spelling of stdClass.

anton-vlasenko committed:
Use factory to create posts.

anton-vlasenko committed:
Add error messages.

anton-vlasenko committed:
Refactor Tests_Editor_DisableContentEditorForNavigationPostType and Tests_Editor_EnableContentEditorForNavigationPostType.
Use data providers to cover edge cases.

anton-vlasenko committed:
Refactor Tests_Editor_DisableBlockEditorForNavigationPostType.
Use data provider to cover edge cases.

costdev commented on tests/phpunit/tests/editor/disableBlockEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,49 @@
+	/**
+	 * @ticket 56266
+	 *
+	 * @dataProvider data_test_it_correctly_handles_different_post_types
+	 */
```
```suggestion
	 * @dataProvider data_test_it_correctly_handles_different_post_types
	 *
	 * @param string $post_type The post type.
	 * @param bool   $value     Whether to disable the block editor.
	 * @param bool   $expected  The expected result.
	 */
```

anton-vlasenko replied:
Fixed in https://github.com/WordPress/wordpress-develop/pull/3005/commits/31788a26830266e44de78e6decb864d6a228e2d0

costdev commented on tests/phpunit/tests/editor/disableBlockEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,49 @@
+	}
+
+	/**
+	 * @ticket 56266
```
```suggestion
	 * Data provider.
	 *
	 * @return array
```

anton-vlasenko replied:
Fixed in https://github.com/WordPress/wordpress-develop/pull/3005/commits/31788a26830266e44de78e6decb864d6a228e2d0

costdev commented on tests/phpunit/tests/editor/disableContentEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,86 @@
+	/**
+	 * @ticket 56266
+	 *
+	 * @dataProvider data_test_it_correctly_handles_different_post_types
+	 */
```
```suggestion
	 * @dataProvider data_test_it_correctly_handles_different_post_types
	 *
	 * @param string $post_type             The post type.
	 * @param bool   $enable_editor_support Whether to enable editor support.
	 * @param bool   $expected              The expected result.
	 * @param string $message               The message to display if the test fails.
	 */
```

anton-vlasenko replied:
Fixed in https://github.com/WordPress/wordpress-develop/pull/3005/commits/31788a26830266e44de78e6decb864d6a228e2d0

costdev commented on tests/phpunit/tests/editor/disableContentEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,86 @@
+	}
+
+	/**
+	 * @ticket 56266
```
```suggestion
	 * Data provider.
	 *
	 * @return array
```

anton-vlasenko replied:
Fixed in https://github.com/WordPress/wordpress-develop/pull/3005/commits/31788a26830266e44de78e6decb864d6a228e2d0

costdev commented on tests/phpunit/tests/editor/enableContentEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,86 @@
+	/**
+	 * @ticket 56266
+	 *
+	 * @dataProvider data_test_it_correctly_handles_different_post_types
+	 */
```
```suggestion
	 * @dataProvider data_test_it_correctly_handles_different_post_types
	 *
	 * @param string $post_type             The post type.
	 * @param bool   $enable_editor_support Whether to enable editor support.
	 * @param bool   $expected              The expected result.
	 * @param string $message               The message to display if the test fails.
	 */
```

anton-vlasenko replied:
Fixed in https://github.com/WordPress/wordpress-develop/pull/3005/commits/31788a26830266e44de78e6decb864d6a228e2d0

costdev commented on tests/phpunit/tests/editor/enableContentEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,86 @@
+	}
+
+	/**
+	 * @ticket 56266
```
```suggestion
	 * Data provider.
	 *
	 * @return array
```

anton-vlasenko replied:
Fixed in https://github.com/WordPress/wordpress-develop/pull/3005/commits/31788a26830266e44de78e6decb864d6a228e2d0

costdev reviewed (changes requested):
Thanks for the updates @anton-vlasenko!

This review adds documentation for test method arguments and makes data provider docblocks consistent with others in the test suite.

anton-vlasenko committed:
Fix DocBlocks.

costdev reviewed (approved):
LGTM! 👍

ironprogrammer commented:
Each of these migrated tests have the same test method name, `test_it_correctly_handles_different_post_types()`, which seems awkward compared with others in the project. Maybe it's just me, but an alternative might be to simplify it to `test_handle_different_post_types()`. Why?

- The inclusion of `_it_` seems unnecessary in the context of these isolated tests, *UNLESS* additional clarity is desired, in which case this should include the function name under test, which seems pretty common.
- We should assume that unit tests already test for "correct" results, making that word redundant. So "incorrect"/error messages for [assertions](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#using-assertions) should be defined in their `message` parameters.
- Like with [git commit messages](https://make.wordpress.org/core/handbook/best-practices/commit-messages/#guidelines), the [imperative voice](https://cbea.ms/git-commit/#imperative) tends to read more easily.

Likewise, [`test_wp_filter_global_styles_post_returns_correct_value()`](https://github.com/WordPress/wordpress-develop/blob/31788a26830266e44de78e6decb864d6a228e2d0/tests/phpunit/tests/kses/wpFilterGlobalStylesPost.php) mentions "correct" in its name, but the test is to "return the **same** value", hence `_return_same_value` sounds closer to the intent.

Of course none of this is critical, and maybe only matters to those who stare at unit tests and terminal output! Just `</two-cents>` 😂

costdev commented:
> @ironprogrammer Each of these migrated tests have the same test method name, test_it_correctly_handles_different_post_types(), which seems awkward compared with others in the project. Maybe it's just me, but an alternative might be to simplify it to test_handle_different_post_types().

I think `test_should_` / `test_should_not_` have precedent as test method prefixes in Core and are certainly consistent with wider unit/integration/e2e testing, making it easier for experienced testers when they begin contributing to Core.

I had a longer message in mind, but as this comment will appear on the Trac ticket, I've instead put this in [a Gist](https://gist.github.com/costdev/bbbdcb112c397174e08b31f357f9696f) to keep the noise down. I'm curious to hear your thoughts on this, so feel free to DM me on Slack to discuss further.

anton-vlasenko committed:
Rename test methods.

anton-vlasenko committed:
Refactor Tests_Kses_WpFilterGlobalStylesPost and add more tests.

anton-vlasenko committed:
Fix the data provider name.

costdev commented on tests/phpunit/tests/editor/disableBlockEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,55 @@
+	 *
+	 * @return array
+	 */
+	public function data_test_should_only_disable_block_editor_for_navigation_post_types() {
```
```suggestion
	public function data_should_only_disable_block_editor_for_navigation_post_types() {
```

anton-vlasenko replied:
Fixed in d6ee08b6441d289e4009cbf66b298f3959fa54ad

costdev commented on tests/phpunit/tests/editor/disableBlockEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,55 @@
+	/**
+	 * @ticket 56266
+	 *
+	 * @dataProvider data_test_should_only_disable_block_editor_for_navigation_post_types
```
```suggestion
	 * @dataProvider data_should_only_disable_block_editor_for_navigation_post_types
```

anton-vlasenko replied:
Fixed in d6ee08b6441d289e4009cbf66b298f3959fa54ad

costdev commented on tests/phpunit/tests/editor/disableContentEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,93 @@
+	/**
+	 * @ticket 56266
+	 *
+	 * @dataProvider data_test_should_only_disable_content_editor_for_navigation_post_types
```
```suggestion
	 * @dataProvider data_should_only_disable_content_editor_for_navigation_post_types
```

anton-vlasenko replied:
Fixed in d6ee08b6441d289e4009cbf66b298f3959fa54ad

costdev commented on tests/phpunit/tests/editor/disableContentEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,93 @@
+	 *
+	 * @return array
+	 */
+	public function data_test_should_only_disable_content_editor_for_navigation_post_types() {
```
```suggestion
	public function data_should_only_disable_content_editor_for_navigation_post_types() {
```

anton-vlasenko replied:
Fixed in d6ee08b6441d289e4009cbf66b298f3959fa54ad

costdev commented on tests/phpunit/tests/editor/enableContentEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,93 @@
+	/**
+	 * @ticket 56266
+	 *
+	 * @dataProvider data_test_should_only_enable_content_editor_for_navigation_post_types
```
```suggestion
	 * @dataProvider data_should_only_enable_content_editor_for_navigation_post_types
```

anton-vlasenko replied:
Fixed in d6ee08b6441d289e4009cbf66b298f3959fa54ad

costdev commented on tests/phpunit/tests/editor/enableContentEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,93 @@
+	 *
+	 * @return array
+	 */
+	public function data_test_should_only_enable_content_editor_for_navigation_post_types() {
```
```suggestion
	public function data_should_only_enable_content_editor_for_navigation_post_types() {
```

anton-vlasenko replied:
Fixed in d6ee08b6441d289e4009cbf66b298f3959fa54ad

costdev reviewed (changes requested):
Very minimal review to remove `test_` from the data provider names. Why?

- `test_` is a prefix for tests.
- `data_` is a prefix for data providers.

There is no need to prefix the data provider with `data_test_`.

`data_test_` exists 51 times in the test suite, versus just `data_...`, which exists 1528 times.

The rest looks good to me! 🙂

anton-vlasenko committed:
Rename data providers. Please see
https://github.com/WordPress/wordpress-develop/pull/3005/files#r948367528
https://github.com/WordPress/wordpress-develop/pull/3005/files#r948367754
https://github.com/WordPress/wordpress-develop/pull/3005/files#r948367838
https://github.com/WordPress/wordpress-develop/pull/3005/files#r948368085

anton-vlasenko commented:
> There is no need to prefix the data provider with `data_test_`.

My bad. Thanks for the review, @costdev!
I've fixed it.

costdev reviewed (approved):
LGTM! :+1:

ironprogrammer commented on tests/phpunit/tests/kses/wpFilterGlobalStylesPost.php:
```diff
@@ -0,0 +1,59 @@
+	 */
+	public function test_should_not_remove_safe_global_style_rules() {
+		$filtered_user_theme_json = $this->filter_global_styles( $this->user_theme_data );
+		$this->assertArrayNotHasKey( 'nonSchemaRule', $filtered_user_theme_json, 'Filtered json data must not contain unsafe global style rules.' );
```
I was thinking Lines 35 and 43 should be swapped:

- The first function, `should_not_remove`, asserts that `nonSchemaRule` was _removed_, which is the opposite of the name.
- Conversely, the other test, `should_remove`, has an assertion that `styles` was _NOT removed_.

This could just be how I'm reading it.

anton-vlasenko replied:
Good catch.
Fixed in f5c52f615ccd2e244206e2cbf65b6e1a59425fef

anton-vlasenko committed:
https://github.com/WordPress/wordpress-develop/pull/3005#discussion_r949660071

robinwpdeveloper commented:
PR looks good to me!  👍

hellofromtonya commented on tests/phpunit/tests/editor/disableContentEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,93 @@
+	 */
+	public function test_should_only_disable_content_editor_for_navigation_post_types( $post_type, $enable_editor_support, $expected, $message ) {
+		$post = $this->create_post( $post_type );
+		if ( $enable_editor_support ) {
+			$this->enable_editor_support();
+			$this->assertTrue( $this->supports_block_editor(), 'Editor support must be enabled before running the test.' );
+		} else {
+			$this->disable_editor_support();
+			$this->assertFalse( $this->supports_block_editor(), 'Editor support must be disabled before running the test.' );
+		}
```
The code here does not need to be retested for each dataset. Rather, it's testing the tests set up of enabling and disabling. That can be done in a separate test methods. In doing, it reduces the code in this test and eliminates the need for a message arg to be passed.

Update coming.

hellofromtonya committed:
Adds more test coverage for editor tests

This commits adds more test coverage to check what happens when passing other types of posts into:
* _disable_block_editor_for_navigation_post_type()
* _disable_content_editor_for_navigation_post_type()
* _enable_content_editor_for_navigation_post_type()

While these functions only work on `'wp_navigation'` post type, the new data sets check all post types.

hellofromtonya commented:
Rebased on top of current `trunk` to ensure the new tests cover any changes without introducing any test leaks to other tests.

anton-vlasenko commented on tests/phpunit/tests/editor/disableContentEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,113 @@
+	 *
+	 * @return array
+	 */
+	public function data_should_not_change_post_type_support() {
+		return array(
+			'post'                => array( 'post' ),
+			'page'                => array( 'page' ),
+			'attachments'         => array( 'attachments' ),
+			'revision'            => array( 'revision' ),
+			'custom_css'          => array( 'custom_css' ),
+			'customize_changeset' => array( 'customize_changeset' ),
+			'nav_menu_item'       => array( 'nav_menu_item' ),
+			'oembed_cache'        => array( 'oembed_cache' ),
+			'user_request'        => array( 'user_request' ),
+			'wp_block'            => array( 'wp_block' ),
+			'wp_template'         => array( 'wp_template' ),
+			'wp_template_part'    => array( 'wp_template_part' ),
+			'wp_global_styles'    => array( 'wp_global_styles' ),
+		);
+	}
```
```suggestion
	public function data_should_not_change_post_type_support() {
		return array(
			'post'                => array( 'post_type' => 'post' ),
			'page'                => array( 'post_type' => 'page' ),
			'attachments'         => array( 'post_type' => 'attachments' ),
			'revision'            => array( 'post_type' => 'revision' ),
			'custom_css'          => array( 'post_type' => 'custom_css' ),
			'customize_changeset' => array( 'post_type' => 'customize_changeset' ),
			'nav_menu_item'       => array( 'post_type' => 'nav_menu_item' ),
			'oembed_cache'        => array( 'post_type' => 'oembed_cache' ),
			'user_request'        => array( 'post_type' => 'user_request' ),
			'wp_block'            => array( 'post_type' => 'wp_block' ),
			'wp_template'         => array( 'post_type' => 'wp_template' ),
			'wp_template_part'    => array( 'post_type' => 'wp_template_part' ),
			'wp_global_styles'    => array( 'post_type' => 'wp_global_styles' ),
		);
	}
```

hellofromtonya replied:
It's fine, but not needed when there's only one piece of data being passed to the function.

hellofromtonya replied:
Looking at other recent tests that Juliette and I did, when there's only 1 data item being passed, adding the parameter name can make the data provider less readable. IMO the standard should be:

* 1 data item => no parameter name
* 2 or more data items => each has it's corresponding parameter name as a key in the dataset

anton-vlasenko replied:
Thank you for the feedback, @hellofromtonya.
I appreciate it.

anton-vlasenko commented on tests/phpunit/tests/editor/disableContentEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,113 @@
+	 *
+	 * @return array
+	 */
+	public function data_should_not_disable() {
+		return array(
+			'post'             => array( 'post' ),
+			'page'             => array( 'page' ),
+			'nav_menu_item'    => array( 'nav_menu_item' ),
+			'oembed_cache'     => array( 'oembed_cache' ),
+			'user_request'     => array( 'user_request' ),
+			'wp_block'         => array( 'wp_block' ),
+			'wp_template'      => array( 'wp_template' ),
+			'wp_template_part' => array( 'wp_template_part' ),
+			'wp_global_styles' => array( 'wp_global_styles' ),
+		);
+	}
```
```suggestion
	public function data_should_not_disable() {
		return array(
			'post'             => array( 'post_type' => 'post' ),
			'page'             => array( 'post_type' => 'page' ),
			'nav_menu_item'    => array( 'post_type' => 'nav_menu_item' ),
			'oembed_cache'     => array( 'post_type' => 'oembed_cache' ),
			'user_request'     => array( 'post_type' => 'user_request' ),
			'wp_block'         => array( 'post_type' => 'wp_block' ),
			'wp_template'      => array( 'post_type' => 'wp_template' ),
			'wp_template_part' => array( 'post_type' => 'wp_template_part' ),
			'wp_global_styles' => array( 'post_type' => 'wp_global_styles' ),
		);
	}
```

anton-vlasenko commented on tests/phpunit/tests/editor/enableContentEditorForNavigationPostType.php:
```diff
@@ -0,0 +1,130 @@
+	 *
+	 * @return array
+	 */
+	public function data_should_not_change_post_type_support() {
+		return array(
+			'post'                => array( 'post' ),
+			'page'                => array( 'page' ),
+			'attachments'         => array( 'attachments' ),
+			'revision'            => array( 'revision' ),
+			'custom_css'          => array( 'custom_css' ),
+			'customize_changeset' => array( 'customize_changeset' ),
+			'nav_menu_item'       => array( 'nav_menu_item' ),
+			'oembed_cache'        => array( 'oembed_cache' ),
+			'user_request'        => array( 'user_request' ),
+			'wp_block'            => array( 'wp_block' ),
+			'wp_template'         => array( 'wp_template' ),
+			'wp_template_part'    => array( 'wp_template_part' ),
+			'wp_global_styles'    => array( 'wp_global_styles' ),
+		);
+	}
```
```suggestion
	public function data_should_not_change_post_type_support() {
		return array(
			'post'                => array( 'post_type' => 'post' ),
			'page'                => array( 'post_type' => 'page' ),
			'attachments'         => array( 'post_type' => 'attachments' ),
			'revision'            => array( 'post_type' => 'revision' ),
			'custom_css'          => array( 'post_type' => 'custom_css' ),
			'customize_changeset' => array( 'post_type' => 'customize_changeset' ),
			'nav_menu_item'       => array( 'post_type' => 'nav_menu_item' ),
			'oembed_cache'        => array( 'post_type' => 'oembed_cache' ),
			'user_request'        => array( 'post_type' => 'user_request' ),
			'wp_block'            => array( 'post_type' => 'wp_block' ),
			'wp_template'         => array( 'post_type' => 'wp_template' ),
			'wp_template_part'    => array( 'post_type' => 'wp_template_part' ),
			'wp_global_styles'    => array( 'post_type' => 'wp_global_styles' ),
		);
	}
```

anton-vlasenko reviewed (commented):
@hellofromtonya What is your opinion on that?

hellofromtonya commented on tests/phpunit/tests/kses/wpFilterGlobalStylesPost.php:
```diff
@@ -0,0 +1,59 @@
+	 *
+	 * @var array
+	 */
+	private $user_theme_data = array(
+		'isGlobalStylesUserThemeJSON' => 1,
+		'version'                     => 1,
+		'nonSchemaRule'               => 'someValue',
+		'styles'                      => array(
+			'blocks' => array(
+				'core/button' => array(
+					'border' => array(
+						'radius' => '0',
+					),
+				),
+			),
+		),
+	);
```
I'm thinking a data provider is needed to have more datasets for better coverage. There are a lot of schema properties to test. What do you all think?

anton-vlasenko replied:
I've added a new dataprovider for the `test_should_not_remove_safe_global_style_rules`.

anton-vlasenko committed:
Commit WIP.

anton-vlasenko committed:
Add tests for verifying that safe rules don't get removed.

anton-vlasenko committed:
Fix CS.

hellofromtonya reviewed (approved):
LGTM 👍 

prepping commit.

hellofromtonya commented:
Committed via changeset https://core.trac.wordpress.org/changeset/54382. Thank you everyone for your contributions!
